### ISSUE TITLE
amp-timeago screen reader verbalization is bad

### ISSUE THREAD

**Issue by TomerAberbach** [ampproject/amphtml#32398]
## What's the issue?

`amp-timeago` is not verbalized well by Voice Over on Mac and I suspect the same is true for other screen readers and other operating systems. There are a few problems:
* The placeholder text is verbalized instead of the generated time ago text and the element is called a "group". Verbalizing the generated time ago text requires an extra key press.
* The element is verbalized separately from sibling text nodes, but it should be merged with them.

## How do we reproduce the issue?

See the [video](https://github.com/ampproject/amphtml/files/5920884/Untitled.mov.zip) of the issue and part of the solution as well as the associated [playground](https://playground.amp.dev/?runtime=amp4email&mode=Responsive#share=PCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoTRlbWFpbCBkYXRhLWNzcy1zdHJpY3Q+CjxoZWFkPgogIDxtZXRhIGNoYXJzZXQ9InV0Zi04Ij4KICA8c2NyaXB0IGFzeW5jIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAuanMiPjwvc2NyaXB0PgogIDxzY3JpcHQgYXN5bmMgY3VzdG9tLWVsZW1lbnQ9ImFtcC10aW1lYWdvIiBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwL2FtcC10aW1lYWdvLTAuMS5qcyI+PC9zY3JpcHQ+CiAgPHN0eWxlIGFtcDRlbWFpbC1ib2lsZXJwbGF0ZT5ib2R5e3Zpc2liaWxpdHk6aGlkZGVufTwvc3R5bGU+CiAgPHN0eWxlIGFtcC1jdXN0b20+CiAgICBib2R5IHsKICAgICAgcGFkZGluZzogMTBweDsKICAgIH0KICAgIHNwYW4gewogICAgICBkaXNwbGF5OiBmbGV4OwogICAgICBmb250LWZhbWlseTogc2Fucy1zZXJpZjsKICAgICAgZm9udC1zaXplOiAxMnB0OwogICAgfQogIDwvc3R5bGU+CjwvaGVhZD4KPGJvZHk+CiAgPGgxPkJhZDwvaDE+CiAgPHNwYW4+CiAgCVNvbWV0aGluZyBoYXBwZW5lZCZuYnNwOwogICAgPGFtcC10aW1lYWdvCiAgICAgIGxheW91dD0iZml4ZWQiCiAgICAgIHdpZHRoPSIxNjAiCiAgICAgIGhlaWdodD0iMjAiCiAgICAgIGRhdGV0aW1lPSIyMDE3LTA0LTExVDAwOjM3OjMzLjgwOVoiCiAgICAgIGxvY2FsZT0iZW4iCiAgICA+CiAgICAgIFNhdHVyZGF5IDExIEFwcmlsIDIwMTcgMDAuMzcKICAgIDwvYW1wLXRpbWVhZ28+CiAgPC9zcGFuPgoKICA8aDE+QmV0dGVyPC9oMT4KICA8c3Bhbj4KICAJU29tZXRoaW5nIGhhcHBlbmVkJm5ic3A7CiAgICA8YW1wLXRpbWVhZ28KICAgICAgbGF5b3V0PSJmaXhlZCIKICAgICAgd2lkdGg9IjE2MCIKICAgICAgaGVpZ2h0PSIyMCIKICAgICAgZGF0ZXRpbWU9IjIwMTctMDQtMTFUMDA6Mzc6MzMuODA5WiIKICAgICAgbG9jYWxlPSJlbiIKICAgICAgcm9sZT0idGV4dCIKICAgID4KICAgICAgU2F0dXJkYXkgMTEgQXByaWwgMjAxNyAwMC4zNwogICAgPC9hbXAtdGltZWFnbz4KICA8L3NwYW4+CjwvYm9keT4KPC9odG1sPgo=).

## What browsers are affected?

All browsers seem to be affected.

## Which AMP version is affected?

The AMP version printed to the console is 2101230412003.

## Proposed Solution

Adding `role="text"` to the `amp-timeago` component, as shown in the video and playground, fixes most of the problems. I think it would be beneficial for `role="text"` to be set on this component by default so that it is accessible by default.

However, even with `role="text"` the placeholder text of the component is still verbalized shortly after the generated time ago text because it is set as the `title` of the component. I question whether the `title` attribute should be set by default for this component. I think it's redundant for screen reader users and I don't think sighted users are hovering over the `amp-timeago` text to see the full datetime. I would suggest not setting the `title` attribute at all. `amp-timeago` should behave like normal text.

**Comment by caroqliu** [comment]
@nainar @zewenliang Would either of you know the how much hover text is relied upon? Could we reasonably "break" this behavior given the placeholder text is not an essential part of the component either before or after it has been replaced with the time ago text?

**Comment by caroqliu** [comment]
Confirmed this only affects version 0.1. Version 1.0 behaves [as expected in the playground](https://playground.amp.dev/amp/undefined#share=PCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoT4KPGhlYWQ+CiAgPG1ldGEgY2hhcnNldD0idXRmLTgiPgogIDx0aXRsZT5hbXAtdGltZWFnbyAtIEV4YW1wbGUgMTwvdGl0bGU+CiAgPHNjcmlwdCBhc3luYyBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwLmpzIj48L3NjcmlwdD4KICA8c2NyaXB0IGN1c3RvbS1lbGVtZW50PSJhbXAtdGltZWFnbyIgc3JjPSJodHRwczovL2Nkbi5hbXBwcm9qZWN0Lm9yZy92MC9hbXAtdGltZWFnby0xLjAuanMiIGFzeW5jPjwvc2NyaXB0PgogIDxsaW5rIHJlbD0iY2Fub25pY2FsIiBocmVmPSJodHRwczovL3ByZXZpZXcuYW1wLmRldi9kb2N1bWVudGF0aW9uL2NvbXBvbmVudHMvYW1wLXRpbWVhZ28tdjEuMC5leGFtcGxlLjEuaHRtbCI+CiAgPG1ldGEgbmFtZT0idmlld3BvcnQiIGNvbnRlbnQ9IndpZHRoPWRldmljZS13aWR0aCI+CiAgPHN0eWxlIGFtcC1ib2lsZXJwbGF0ZT5ib2R5ey13ZWJraXQtYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7LW1vei1hbmltYXRpb246LWFtcC1zdGFydCA4cyBzdGVwcygxLGVuZCkgMHMgMSBub3JtYWwgYm90aDstbXMtYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7YW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGh9QC13ZWJraXQta2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fUAtbW96LWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW1zLWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW8ta2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fUBrZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19PC9zdHlsZT48bm9zY3JpcHQ+PHN0eWxlIGFtcC1ib2lsZXJwbGF0ZT5ib2R5ey13ZWJraXQtYW5pbWF0aW9uOm5vbmU7LW1vei1hbmltYXRpb246bm9uZTstbXMtYW5pbWF0aW9uOm5vbmU7YW5pbWF0aW9uOm5vbmV9PC9zdHlsZT48L25vc2NyaXB0PgogIDxzdHlsZSBhbXAtY3VzdG9tPgogICAgYm9keSB7CiAgICAgIHBhZGRpbmc6IDEwcHg7CiAgICB9CiAgICBzcGFuIHsKICAgICAgZGlzcGxheTogZmxleDsKICAgICAgZm9udC1mYW1pbHk6IHNhbnMtc2VyaWY7CiAgICAgIGZvbnQtc2l6ZTogMTJwdDsKICAgIH0KICA8L3N0eWxlPgogIDwvaGVhZD4KICA8Ym9keT4KICAgIDxoMT5CYWQ8L2gxPgogICAgPHNwYW4+CiAgICAgIFNvbWV0aGluZyBoYXBwZW5lZCZuYnNwOwogICAgICA8YW1wLXRpbWVhZ28KICAgICAgICBsYXlvdXQ9ImZpeGVkIgogICAgICAgIHdpZHRoPSIxNjAiCiAgICAgICAgaGVpZ2h0PSIyMCIKICAgICAgICBkYXRldGltZT0iMjAxNy0wNC0xMVQwMDozNzozMy44MDlaIgogICAgICAgIGxvY2FsZT0iZW4iCiAgICAgID4KICAgICAgICBTYXR1cmRheSAxMSBBcHJpbCAyMDE3IDAwLjM3CiAgICAgIDwvYW1wLXRpbWVhZ28+CiAgICA8L3NwYW4+CgogICAgPGgxPkJldHRlcjwvaDE+CiAgICA8c3Bhbj4KICAgICAgU29tZXRoaW5nIGhhcHBlbmVkJm5ic3A7CiAgICAgIDxhbXAtdGltZWFnbwogICAgICAgIGxheW91dD0iZml4ZWQiCiAgICAgICAgd2lkdGg9IjE2MCIKICAgICAgICBoZWlnaHQ9IjIwIgogICAgICAgIGRhdGV0aW1lPSIyMDE3LTA0LTExVDAwOjM3OjMzLjgwOVoiCiAgICAgICAgbG9jYWxlPSJlbiIKICAgICAgICByb2xlPSJ0ZXh0IgogICAgICA+CiAgICAgICAgU2F0dXJkYXkgMTEgQXByaWwgMjAxNyAwMC4zNwogICAgICA8L2FtcC10aW1lYWdvPgogICAgPC9zcGFuPgogIDwvYm9keT4KPC9odG1sPgo=).

**Comment by kristoferbaxter** [comment]
Thanks @caroqliu, closing this issue with the fix for `0.1` landed.
